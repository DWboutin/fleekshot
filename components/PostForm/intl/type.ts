export interface PostFormIntl {
  postForm: {
    uploadImage: string;
    share: string;
    inputLabel: string;
    errors: {
      tooShort: string;
      tooLong: string;
      isRequired: string;
      imageIsRequired: string;
    };
  };
}

export enum PostFormIntlId {
  postForm_uploadImage = "postForm_uploadImage",
  postForm_share = "postForm_share",
  postForm_inputLabel = "postForm_inputLabel",
  postForm_errors_tooShort = "postForm_errors_tooShort",
  postForm_errors_tooLong = "postForm_errors_tooLong",
  postForm_errors_isRequired = "postForm_errors_isRequired",
  postForm_errors_imageIsRequired = "postForm_errors_imageIsRequired",
}

export interface PostFormIntlFlatten extends Record<PostFormIntlId, string> {}
