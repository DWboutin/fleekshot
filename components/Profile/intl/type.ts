export interface ProfileIntl {
  profile: {
    title: string;
    uploadPicture: string;
  };
}

export enum ProfileIntlId {
  profile_title = "profile_title",
  profile_uploadPicture = "profile_uploadPicture",
}

export interface ProfileIntlFlatten extends Record<ProfileIntlId, string> {}
