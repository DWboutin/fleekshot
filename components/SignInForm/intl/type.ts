export interface SignInFormIntl {
  signInForm: {
    input: {
      username: {
        label: string;
      };
      password: {
        label: string;
      };
    };
    button: {
      submit: string;
    };
    errors: {
      tooShort: string;
      tooLong: string;
      isRequired: string;
      notStrongEnough: string;
      mismatch: string;
    };
  };
}

export enum SignInFormIntlId {
  signInForm_input_username_label = "signInForm_input_username_label",
  signInForm_input_password_label = "signInForm_input_password_label",
  signInForm_button_submit = "signInForm_button_submit",
  signInForm_errors_tooShort = "signInForm_errors_tooShort",
  signInForm_errors_tooLong = "signInForm_errors_tooLong",
  signInForm_errors_isRequired = "signInForm_errors_isRequired",
  signInForm_errors_notStrongEnough = "signInForm_errors_notStrongEnough",
  signInForm_errors_mismatch = "signInForm_errors_mismatch",
}

export interface SignInFormIntlFlatten
  extends Record<SignInFormIntlId, string> {}
