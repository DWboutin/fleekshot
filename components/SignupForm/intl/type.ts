export interface SignUpFormIntl {
  signUpForm: {
    title: string;
    input: {
      name: {
        label: string;
      };
      username: {
        label: string;
      };
      password: {
        label: string;
      };
      confirmPassword: {
        label: string;
      };
    };
    button: {
      submit: string;
    };
    errors: {
      tooShort: string;
      tooLong: string;
      isRequired: string;
      notStrongEnough: string;
      mismatch: string;
    };
  };
}

export enum SignUpFormIntlId {
  signUpForm_title = "signUpForm_title",
  signUpForm_input_name_label = "signUpForm_input_name_label",
  signUpForm_input_username_label = "signUpForm_input_username_label",
  signUpForm_input_password_label = "signUpForm_input_password_label",
  signUpForm_input_confirmPassword_label = "signUpForm_input_confirmPassword_label",
  signUpForm_button_submit = "signUpForm_button_submit",
  signUpForm_errors_tooShort = "signUpForm_errors_tooShort",
  signUpForm_errors_tooLong = "signUpForm_errors_tooLong",
  signUpForm_errors_isRequired = "signUpForm_errors_isRequired",
  signUpForm_errors_notStrongEnough = "signUpForm_errors_notStrongEnough",
  signUpForm_errors_mismatch = "signUpForm_errors_mismatch",
}

export interface SignUpFormIntlFlatten
  extends Record<SignUpFormIntlId, string> {}
