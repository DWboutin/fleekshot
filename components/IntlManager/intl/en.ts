import flat from "flat";

import signUpForm from "../../SignUpForm/intl/en";
import signInForm from "../../SignInForm/intl/en";
import { IntlMessages } from "./types";

const en = {
  hello: "Hello",
  ...signUpForm,
  ...signInForm,
};

export default flat(en, { delimiter: "_" }) as IntlMessages;
