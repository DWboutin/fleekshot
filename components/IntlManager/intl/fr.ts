import flat from "flat";

import signUpForm from "../../SignUpForm/intl/fr";
import signInForm from "../../SignInForm/intl/fr";
import profile from "../../Profile/intl/fr";
import { IntlMessages } from "./types";

const fr = {
  home: "Accueil",
  ...signUpForm,
  ...signInForm,
  ...profile,
};

export default flat(fr, { delimiter: "_" }) as IntlMessages;
