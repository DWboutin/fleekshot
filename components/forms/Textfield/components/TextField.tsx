import React, { useEffect, useRef } from "react";
import { useIntl } from "react-intl";
import styled from "styled-components";
import { ThemeContainer } from "../../../../styles/styles";
import { toREM } from "../../../../styles/typography";
import Boxicon from "../../../BoxIcons/BoxIcons";
import { useTextField } from "../hooks/useTextField";

interface ContainerProps extends ThemeContainer {
  isFocused: boolean;
  hasValue: boolean;
}

export type TextFieldTypes = "email" | "password" | "text";

export const Container = styled.div<ContainerProps>`
  display: flex;
  flex: 1;
  flex-direction: row;
  position: relative;
  height: 36px;
  ${({ theme }: ThemeContainer) => `
    background-color: ${theme.forms.textField.bg};
    border: 1px solid ${theme.forms.textField.border};
    border-radius: ${theme.forms.textField.borderRadius};
  `}

  ${({ isFocused, theme }: ContainerProps) =>
    isFocused &&
    `
      border: 1px solid ${theme.forms.textField.borderFocus};
    `}

  &>label {
    display: flex;
    flex: 1;

    & > span:first-child {
      transform-origin: left;
      transition: transform ease-out 0.1s, -webkit-transform ease-out 0.1s;
      z-index: 3;

      ${({ isFocused, hasValue }: ContainerProps) =>
        (isFocused || hasValue) &&
        `
        transform: scale(0.7) translateY(-14px);
      `}
    }
  }
`;

const LabelContainer = styled.span`
  display: inline-block;
  position: absolute;
  left: 8px;
  height: 100%;
  cursor: text;
  height: 36px;
  line-height: 36px;
  ${({ theme }: ThemeContainer) => `
    color: ${theme.forms.textField.label};
  `}
  z-index: 1;
`;

const Input = styled.input`
  flex: 1;
  padding: 12px 8px 4px;
  font-size: ${toREM(12)};
  height: 36px;
  border: 0px;
  outline: none;
  background: transparent;
  cursor: text;
  z-index: 2;
`;

const PasswordToggle = styled.button`
  display: flex;
  align-items: center;
  border: 0;
  background: transparent;
  padding: 5px 5px 3px;
  cursor: pointer;
  color: ${({ theme }: ThemeContainer) => theme.forms.textField.label};
`;

const ErrorIcon = styled.span`
  display: flex;
  align-items: center;
  border: 0;
  background: transparent;
  padding: 5px 5px 3px;
  color: ${({ theme }: ThemeContainer) => theme.forms.errorColor};
`;

export interface Props {
  id: string;
  name: string;
  type: TextFieldTypes;
  label: string;
  value?: string;
  defaultValue?: string;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  error?: string | false;
}

const TextField: React.VoidFunctionComponent<Props> = ({
  id,
  name,
  type,
  label,
  value,
  defaultValue,
  onChange,
  error,
}) => {
  const intl = useIntl();
  const {
    selectors: { fieldValue, fieldType, isFocused },
    actions: {
      handleOnChange,
      handleOnBlur,
      handleOnFocus,
      handlePasswordToggle,
    },
  } = useTextField(type, defaultValue);

  return (
    <Container isFocused={isFocused} hasValue={!!value}>
      <label htmlFor={id}>
        <LabelContainer>{label}</LabelContainer>
        <Input
          id={id}
          name={name}
          type={fieldType}
          value={value || fieldValue}
          onChange={onChange || handleOnChange}
          onFocus={handleOnFocus}
          onBlur={handleOnBlur}
          autoComplete="off"
        />
        {!!error && (
          <ErrorIcon title={intl.formatMessage({ id: error })}>
            <Boxicon name="x-circle" size="small" />
          </ErrorIcon>
        )}
        {type === "password" && (
          <PasswordToggle type="button" onClick={handlePasswordToggle}>
            <Boxicon
              name={fieldType === "password" ? "hide" : "show"}
              size="small"
            />
          </PasswordToggle>
        )}
      </label>
    </Container>
  );
};

export default TextField;
