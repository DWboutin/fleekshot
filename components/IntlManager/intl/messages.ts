import en from "./en";
import fr from "./fr";

export enum IntlLocale {
  en = "en",
  fr = "fr",
}

export interface IntlMessages extends Record<string, string> {
  hello: string;
}

const messages: Record<IntlLocale, IntlMessages> = {
  en,
  fr,
};

export default messages;
