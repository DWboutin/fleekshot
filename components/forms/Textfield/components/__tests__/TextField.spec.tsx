import {
  fireEvent,
  render,
  RenderResult,
  screen,
} from "@testing-library/react";
import TestWrapper from "../../../../../utils/TestWrapper";
import { SignInFormIntlId } from "../../../../SignInForm/intl/type";

import TextField, { Props } from "../TextField";

describe("<TextField />", () => {
  let renderResult: RenderResult;

  const renderTextField = ({
    id,
    name,
    type,
    label,
    defaultValue,
    error,
  }: Props) => {
    renderResult = render(
      <TestWrapper>
        <TextField
          id={id}
          name={name}
          type={type}
          label={label}
          defaultValue={defaultValue}
          error={error}
        />
      </TestWrapper>
    );
  };

  describe("default rendering", () => {
    const ID = "id";
    const NAME = "name";
    const TYPE = "text";
    const LABEL = "field label";

    beforeEach(() => {
      renderTextField({
        id: ID,
        name: NAME,
        type: TYPE,
        label: LABEL,
      });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });

    it(`should contain the label "${LABEL}"`, () => {
      expect(screen.getByText(LABEL)).toBeInTheDocument();
    });

    it(`should not contain the toggle button`, () => {
      expect(screen.queryByRole("button")).not.toBeInTheDocument();
    });

    it("should contain the input with good attributes", () => {
      const input = renderResult.container.querySelector("input");

      expect(screen.getByRole("textbox")).toBeInTheDocument();
      expect(input?.getAttribute("id")).toBe(ID);
      expect(input?.getAttribute("name")).toBe(NAME);
      expect(input?.getAttribute("type")).toBe(TYPE);
    });
  });

  describe('type="password"', () => {
    const ID = "id";
    const NAME = "name";
    const TYPE = "password";
    const LABEL = "field label";

    beforeEach(() => {
      renderTextField({
        id: ID,
        name: NAME,
        type: TYPE,
        label: LABEL,
      });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });

    it(`should contain the label "${LABEL}"`, () => {
      expect(screen.getByText(LABEL)).toBeInTheDocument();
    });

    it(`should contain the toggle button`, () => {
      expect(screen.getByRole("button")).toBeInTheDocument();
    });

    it("should contain the input with good attributes", () => {
      const input = renderResult.container.querySelector("input");

      expect(input).toBeInTheDocument();
      expect(input?.getAttribute("id")).toBe(ID);
      expect(input?.getAttribute("name")).toBe(NAME);
      expect(input?.getAttribute("type")).toBe(TYPE);
    });

    it("should toggle input type crrectly when clicking on toggle button", () => {
      const input = renderResult.container.querySelector("input");

      expect(input?.getAttribute("type")).toBe(TYPE);

      fireEvent.click(screen.getByRole("button"));

      expect(input?.getAttribute("type")).toBe("text");

      fireEvent.click(screen.getByRole("button"));

      expect(input?.getAttribute("type")).toBe(TYPE);
    });
  });

  describe("with error", () => {
    const ID = "id";
    const NAME = "name";
    const TYPE = "text";
    const LABEL = "field label";
    const ERROR = SignInFormIntlId.signInForm_errors_isRequired;

    beforeEach(() => {
      renderTextField({
        id: ID,
        name: NAME,
        type: TYPE,
        label: LABEL,
        error: ERROR,
      });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });

    it(`should contain the error icon`, () => {
      const errorIconContainer =
        renderResult.container.querySelector("span:last-child");
      const errorIcon =
        renderResult.container.querySelector("span:last-child i");

      expect(errorIconContainer).toBeInTheDocument();
      expect(errorIcon).toBeInTheDocument();
      expect(errorIcon?.classList).toContain("bx-x-circle");
    });
  });
});
