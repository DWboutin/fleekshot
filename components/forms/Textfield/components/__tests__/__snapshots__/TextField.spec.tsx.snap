// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> default rendering should match snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL HCHGw"
  >
    <label
      for="id"
    >
      <span
        class="sc-gsDKAQ Nzkep"
      >
        field label
      </span>
      <input
        class="sc-dkPtRN wkokf"
        id="id"
        name="name"
        type="text"
        value=""
      />
    </label>
  </div>
</div>
`;

exports[`<TextField /> type="password" should match snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL HCHGw"
  >
    <label
      for="id"
    >
      <span
        class="sc-gsDKAQ Nzkep"
      >
        field label
      </span>
      <input
        class="sc-dkPtRN wkokf"
        id="id"
        name="name"
        type="password"
        value=""
      />
      <button
        class="sc-hKwDye fihsgk"
        type="button"
      >
        <i
          class="bx bx-hide bx-sm"
        />
      </button>
    </label>
  </div>
</div>
`;

exports[`<TextField /> with error should match snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL HCHGw"
  >
    <label
      for="id"
    >
      <span
        class="sc-gsDKAQ Nzkep"
      >
        field label
      </span>
      <input
        class="sc-dkPtRN wkokf"
        id="id"
        name="name"
        type="text"
        value=""
      />
      <span
        class="sc-eCImPb jjtWCt"
        title="error message"
      >
        <i
          class="bx bx-x-circle bx-sm"
        />
      </span>
    </label>
  </div>
</div>
`;
